### PR TITLE
[sc-4211] Documentation writing for new Editor

### PR DESCRIPTION
**Pull request by Bondok6**
In this PR:

I added an info/help icon to accordions, tabs, and decision-trees features

[Card](https://app.shortcut.com/helpjuice/story/4211/documentation-writing-for-new-editor)

> It's on staging feel free to test it there

helpjuice PR: https://github.com/hajrice/helpjuice4/pull/3138

[Screencast from 03 أغس, 2023 EEST 06:44:31 م.webm](https://github.com/Helpjuice/ckeditor5/assets/45231276/a65118cb-68ef-41ff-bf0d-faff70b01397)

### CODE REVIEW

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/accordion/styles.css [hunk 1]
```diff
@@ -131,3 +131,16 @@
...
+
+.helpjuice-accordion-info {
+	display: block;
+	position: absolute;
+	bottom: 20px;
+    right: 25px;
+	height: 16px;
+    width: 16px;
+    border-radius: 50%;
+	background-color: #FFFFFF;
```
Comment (SpaYco) [inline packages/ckeditor5-build-helpjuice-v2/src/plugins/accordion/styles.css]: i would make the color match the close button instead.

**Review by SpaYco — APPROVED** [review]
nice work!! i left one comment :)

**Review by NicholasJarr — COMMENTED** [review]
@Bondok6 Nice work 👍🏻. I tested and it worked without issues. I only have a suggestion: your implementation adds a new model (`accordionInfo`) and this will be persisted to the article and sent to the frontend, right?

What if we did something similar to the `EmbeddedIFrame` feature?

https://github.com/Helpjuice/ckeditor5/blob/d166594c45e1a1286c943b696430b192fcfed036/packages/ckeditor5-build-helpjuice-v2/src/plugins/embeddediframe/embeddediframeediting.js#L51-L84

The `editingDowncast` adds extra elements that are only visible on the editor and not included in the `dataDowncast` (so they aren't persisted to the article itself). This could be implemented in the accordions for example, by removing the `accordionInfo` and changing the `accordion` `editingDowncast` to something like this:

```js
conversion.for("editingDowncast").elementToElement({
    model: "accordion",
    view: (modelElement, { writer: viewWriter }) => {
        const link = viewWriter.createAttributeElement("a", {
            class: 'helpjuice-accordion-info',
            href: 'https://help.helpjuice.com/how-to-insert-accordions',
            target: '_blank',
            title: 'Read more about accordions'
        });
        const div = viewWriter.createContainerElement("div", {
	    class: "helpjuice-accordion",
            "data-controller": modelElement.getAttribute("data-controller")
        }, [ link ]);

         return toWidget(div, viewWriter);
    }
});
```

I'm not 100% sure that it will work (CKEditor gets weird sometimes), but if it does, I think it's a better approach since there will be no changes to the schema. If after testing you notice that it's not working or if it will take too long, just let me know and we can keep your implementation.

Any thoughts?

**Comment by Bondok6** [comment]
Hey @NicholasJarr,  that's a good point, thanks for the suggestion
Yeh, I think this will work, let me try it and see

**Comment by Bondok6** [comment]
@NicholasJarr and @SpaYco thanks for the review,

I applied Nicholas's suggestion and I think it works well

Just wondering, if Should we use `createAttributeElement` or `createUIElement`, Not sure what the difference is but I've tried both and they work the same.

> Also I changed the design, added the link and the help icon on the top

image

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/accordion/accordionediting.js [hunk 2]
```diff
@@ -77,10 +77,16 @@ export default class AccordionEditing extends Plugin {
...
 			view: (modelElement, { writer: viewWriter }) => {
+				const link = viewWriter.createAttributeElement("a", {
+					class: 'helpjuice-accordion-info',
+					href: 'https://help.helpjuice.com/how-to-insert-accordions',
+					target: '_blank',
+					title: 'Read more about accordions'
+				});
 				const div = viewWriter.createContainerElement("div", {
-					class: "helpjuice-accordion",
+				class: "helpjuice-accordion",
```
Comment (NicholasJarr) [inline packages/ckeditor5-build-helpjuice-v2/src/plugins/accordion/accordionediting.js]: Can you fix the formatting here?

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/tab/tabediting.js [hunk 3]
```diff
@@ -84,10 +84,16 @@ export default class TabEditing extends Plugin {
 		conversion.for("editingDowncast").elementToElement({
 			model: "tab",
 			view: (modelElement, { writer: viewWriter }) => {
+				const link = viewWriter.createAttributeElement("a", {
+					class: 'helpjuice-tab-info',
+					href: 'https://help.helpjuice.com/how-to-insert-tabs',
```
Comment (NicholasJarr) [inline packages/ckeditor5-build-helpjuice-v2/src/plugins/tab/tabediting.js:89]: I noticed that this article on HJ is on the wrong category:

image

Can you update the article's category?

**Review by NicholasJarr — COMMENTED** [review]
Nice 👍🏻 

> Just wondering, Should we use `createAttributeElement` or `createUIElement`, Not sure what the difference is but I've tried both and they work the same.

You can go with the `createUIElement`. I didn't know about it. CKEditor docs for reference: https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_view_uielement-UIElement.html.

BTW, can we make the info icon smaller? @SpaYco any thoughts on this?

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/accordion/accordionediting.js (hunk 2 above)
Comment (Bondok6) [reply packages/ckeditor5-build-helpjuice-v2/src/plugins/accordion/accordionediting.js]: sure, fixed

**Comment by Bondok6** [comment]
> Nice 👍🏻
> 
> > Just wondering, Should we use `createAttributeElement` or `createUIElement`, Not sure what the difference is but I've tried both and they work the same.
> 
> You can go with the `createUIElement`. I didn't know about it. CKEditor docs for reference: https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_view_uielement-UIElement.html.
> 

Done

> BTW, can we make the info icon smaller? @SpaYco any thoughts on this?

Okay, done 👍🏻

File: packages/ckeditor5-build-helpjuice-v2/src/plugins/tab/tabediting.js (hunk 3 above)
Comment (Bondok6) [reply packages/ckeditor5-build-helpjuice-v2/src/plugins/tab/tabediting.js:89]: Sure, I'll make sure to update it. 
Also, I will update the content of all articles before deploying.

**Review by NicholasJarr — APPROVED** [review]

**Pull request merged**